### ISSUE TITLE
[FEATURE REQUEST] Improve WireGuard to allow for on-demand usage

### ISSUE THREAD

**Issue by FlechazoPh** [SideStore/SideStore#215]
1 Can we use other vpn apps like quantumult x to use vpn , replace wireguard ,because I run it background all the day, so I didn't need to close my vpn and open the wireguard ,and reopen it .

2 Or, Can we use IOS shortcut to auto open wireguard and open vpn toggle,  auto refresh the apps , and close it

**Comment by CangWeiohh** [comment]
Article 2 the wireguard software developers needed to pull requests.

**Comment by FlechazoPh** [comment]
> The second need feedback to wireguard software developers

And sidestore need a shortcut to auto refresh too.

**Comment by Spidy123222** [comment]
> > The second need feedback to wireguard software developers
> 
> 
> 
> And sidestore need a shortcut to auto refresh too.

Sidestore already has background refresh. Just other things prevent it working properly.

**Comment by lanceuppercut47** [comment]
> Sidestore already has background refresh. Just other things prevent it working properly. But using shortcuts thing is possible using passpartout and paying for their app and making a shortcut to enable and disable vpn and just make sidestore refresh via shortcut.

Is it possible to do all this when the device is locked (so I can set it to run overnight) or do I need the device to be unlocked at time of running?

**Comment by Spidy123222** [comment]
I actually haven't payed attention to that but I would assume you need to be unlocked since anytime you try to sideload or do jit it requires it to be unlocked. At least this can happen when your doing something else on the device.

**Comment by lanceuppercut47** [comment]
> I actually haven't payed attention to that but I would assume you need to be unlocked since anytime you try to sideload or do jit it requires it to be unlocked. At least this can happen when your doing something else on the device.

True, but with AltStore/SideStore I can set it to refresh in the background and whilst the device is locked and it won't interrupt what I'm doing if it's unlocked.

The easy way out is to leave the VPN on all the time and let SideStore automatically refresh when it needs to, but I just can't bring myself to do that.

**Comment by Spidy123222** [comment]
Yeah that makes sense if it works that way. The vpn is just on device meaning your idevice hosts it which also means no internet goes through it which means no battery drain from it if that's your worry. It's equivalent of pinging 127.0.0.1.

**Comment by lanceuppercut47** [comment]
> Yeah that makes sense if it works that way. The vpn is just on device meaning your idevice hosts it which also means no internet goes through it which means no battery drain from it if that's your worry. It's equivalent of pinging 127.0.0.1. 

Oh right, I wasn't really aware of how this VPN is set up, so it only does anything when I'm refreshing apps, all the other time it does nothing at all even though its enabled/active?

**Comment by Spidy123222** [comment]
> > Yeah that makes sense if it works that way. The vpn is just on device meaning your idevice hosts it which also means no internet goes through it which means no battery drain from it if that's your worry. It's equivalent of pinging 127.0.0.1. 
> 
> Oh right, I wasn't really aware of how this VPN is set up, so it only does anything when I'm refreshing apps, all the other time it does nothing at all even though its enabled/active?

Yeah it doesn't do anything in background other than exist to function for sidestore. If your using app it's needed and if you are outside of app it will try to refresh in background which needs vpn on for it and in future hopefully have auto enable jit for apps once stuff is more stable.

**Comment by junepark678** [comment]
this is sadly out of our scope

**Comment by mahee96** [comment]
wireguard is retired in 0.6.2+ so, this won't be pursued.
